tmva/sofie/test/TestGemmDerivative.cxx

```diff
@@ -0,0 +1,115 @@
+#include <TInterpreter.h>
+#include <TInterpreterValue.h>
+
+#include "gtest/gtest.h"
+
+#include <stdexcept>
+
+class GemmTest : public testing::TestWithParam<std::tuple<int, int, int>> {
+public:
+   void SetUp() override
+   {
+      m = std::get<0>(GetParam());
+      n = std::get<1>(GetParam());
+      k = std::get<2>(GetParam());
+   }
+
+protected:
+   int m;
+   int n;
+   int k;
+};
+
+void validator(float *variables, int m, int n, int k, float *target)
+{
+   for (int i = 0; i < m * k + k * n + 1; ++i) {
+      target[i] = 0;
+      if (i == 0) {
+         // Check derivative with alpha
+         int _a;
+         int _b;
+         for (int _m = 1; _m < m + 1; ++_m) {
+            for (int _n = 1; _n < n + 1; ++_n) {
+               for (int _k = 1; _k < k + 1; ++_k) {
+                  _a = (_k - 1) * m + (_m - 1) % m + 1;
+                  _b = (_n - 1) * k + (_k - 1) % k + 1;
+                  target[i] += variables[_a] * variables[_b + m * k];
+               }
+            }
+         }
+      } else if ((i - 1) < m * k) {
+         // Get current position in m, k
+         int _m = (i - 1) % m + 1;
+         int _k = ((i - 1) - _m + 1) / m + 1;
+         for (int _n = 1; _n < n + 1; ++_n) {
+            int index = m * k + (_k - 1) % k + (_n - 1) * k;
+            target[i] += variables[index + 1];
+         }
+      } else {
+         int _k = ((i - 1) - m * k) % k + 1;
+         for (int _m = 1; _m < m + 1; ++_m) {
+            int index = (_m - 1) % m + (_k - 1) * m;
+            target[i] += variables[index + 1];
+         }
+      }
+   }
+}
+
+TEST_P(GemmTest, GemmTestDerivative)
+{
+   float variables[] = {1., 3., 4., 2., 2., 4., 0., 6., 1., 2., 5., 7., 3., 1., 4., 2.};
+   float grad_output[] = {0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.};
+   float target[] = {0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.};
+
+   static bool declared = false;
+   if (declared == false) {
+      gInterpreter->Declare(R"cpp(
+            #include <Math/CladDerivator.h>
+
+            float gemm_function(float *variables, int m, int n, int k) {
+                // variable is assumed to pack Amk and Bkn,
+                // so it's of length m*k + k*n
+
+                float alpha = variables[0];
+                float *matA = &variables[1];
+                float *matB = &variables[m*k+1];
+
+                constexpr int n_out = 16; // Allocate a maximum length of arrays here
+                assert (m*n+1 <= n_out);
+
+                float output[n_out];
+                float matC[n_out]{};
```

Comment:
```suggestion
                float matC[n_out]{1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.};
```
If we make `matC` non-zero, the derivative with respect to `beta` will be non-zero and we can also validate it by adding `beta` to the variables (the reference result would be the sum of all elements of C).